src/timepicker/timepicker.js

```diff
@@ -254,9 +312,14 @@ angular.module('ui.bootstrap.timepicker', [])
     $scope.meridian = selected.getHours() < 12 ? meridians[0] : meridians[1];
   }
 
-  function addMinutes( minutes ) {
-    var dt = new Date( selected.getTime() + minutes * 60000 );
-    selected.setHours( dt.getHours(), dt.getMinutes() );
+  function addMinutes( date,  minutes ) {
+    var dt = new Date( date.getTime() + minutes * 60000 );
+    var newDate = new Date( date.getTime() ).setHours( dt.getHours(), dt.getMinutes() );
+    return new Date( newDate );
```

Comment:
Oh, I see now.

It would likely be more efficient to do
```
var newDate = new Date( date.getTime() + minutes * 60000 );
newDate.setHours( dt.getHours(), dt.getMinutes() );
return newDate;
```
The `dt` date existed as a temporary structure to figure out how to modify the existing `selected` date, but now we can just return the date of interest